l3kernel/l3color.dtx

```diff
@@ -1377,6 +1385,75 @@
 %    \end{macrocode}
 % \end{macro}
 %
+% \begin{macro}[EXP] {
+%     \__color_parse_model_lms:nnn, \__color_parse_model_cbrtlms:nnn,
+%     \__color_parse_model_oklab:nnn, \__color_parse_model_oklab:w
+%   }
+%   The Oklab color space is defined by first transforming the color values to
+%   the responses of the L, M and S cones of a standard human observer, then
+%   taking those responses to the power of $\frac{1}{3}$, and finally applying
+%   one more matrix transformation to obtain the $l$, $a$, and $b$ values.
+%   Here, we just go the other way round.
+%    \begin{macrocode}
+\cs_new:Nn \__color_parse_model_lms:nnn
+  {
+    \__color_parse_model_linearrgb:nnn
+      { \fp_eval:n {  4.0767416621 * #1 - 3.3077115913 * #2 + 0.2309699292 * #3 } }
+      { \fp_eval:n { -1.2684380046 * #1 + 2.6097574011 * #2 - 0.3413193965 * #3 } }
+      { \fp_eval:n { -0.0041960863 * #1 - 0.7034186147 * #2 + 1.7076147010 * #3 } }
+  }
+\cs_new:Nn \__color_parse_model_cbrtlms:nnn
+  {
+    \__color_parse_model_lms:nnn
+      { \fp_eval:n { #1 ^ 3 } }
+      { \fp_eval:n { #2 ^ 3 } }
+      { \fp_eval:n { #3 ^ 3 } }
+  }
+\cs_new:Nn \__color_parse_model_oklab:nnn
+  {
+    \__color_parse_model_cbrtlms:nnn
+      { \fp_eval:n { #1 + 0.3963377774 * #2 + 0.2158037573 * #3 } }
+      { \fp_eval:n { #1 - 0.1055613458 * #2 - 0.0638541728 * #3 } }
+      { \fp_eval:n { #1 - 0.0894841775 * #2 - 1.2914855480 * #3 } }
+  }
+\cs_new:Npn \__color_parse_model_oklab:w #1 , #2 , #3 , #4 \s__color_stop
+  { \__color_parse_model_oklab:nnn { #1 } { #2 } { #3 } }
+%    \end{macrocode}
+% \end{macro}
+% \begin{macro}[EXP]
+%   { \__color_parse_model_linearrgb:nnn, \__color_parse_model_linearrgb_aux:n }
+%   Encode the tristimulus values of the sRGB primaries according to sRGB.
+%    \begin{macrocode}
+\cs_new:Nn \__color_parse_model_linearrgb:nnn
+  {
+    { rgb }
+    {
+      \__color_parse_model_linearrgb_aux:n { #1 } ~
+      \__color_parse_model_linearrgb_aux:n { #2 } ~
+      \__color_parse_model_linearrgb_aux:n { #3 }
+    }
+  }
+\cs_new:Nn \__color_parse_model_linearrgb_aux:n
+  {
+    \fp_compare:nNnTF { #1 } < { 0.0031308 }
+      { \fp_eval:n { 12.92 * max ( 0 , #1 ) } }
+      { \fp_eval:n { 1.055 * min ( 1 , #1 ) ^ (1/2.4) - 0.055 } }
```

Comment:
Division is 'expensive' - can we pre-calculate `1/2.4`?

Reply:
I am going with 0.4166666666666667 now.